### PR TITLE
SSH management panel

### PR DESCRIPTION
**Pull request by timstanley1985**
Still a work in progress but i would like to add SSH configuration options to the front end.

- [x]  Change SSH port
- [x] Add authorized_key
- [x] Add option to disable password login

### CODE REVIEW

**Comment by timstanley1985** [comment]
I am having problems with the ssh_key, it keeps substituting spaces with + symbols.  Ant thoughts on how I could get around this?

**Comment by jmtatsch** [comment]
Can you point out on which line the problem with the spaces occur?

File: firmware_mod/www/system.html [hunk 1]
```diff
@@ -66,6 +69,23 @@ <h1> System configuration</h1>
         <label for="usb_eth">Enable USB Ethernet (Disables Wifi)</label>
   </div>
 
+  <div id="ssh" class="w3-container tab" style="display:none">
+        <p></p>
+        <label>SSH Authorized Keys</label>
+        <input id="ssh_key" class="w3-input">
```
Comment (timstanley1985) [inline firmware_mod/www/system.html:75]: If I enter text here, for example "my test key"

File: firmware_mod/www/cgi-bin/ui_system.cgi [hunk 2]
```diff
@@ -78,6 +81,16 @@ if [ -n "$F_cmd" ]; then
 		echo "<p>Setting wifi password to: $wifi_password</p>"
 		wpa_config_set psk "\"$wifi_password\""
 	fi
+  if [ -n ${F_ssh_port} ]; then
+		ssh_port=$(printf '%b' "${F_ssh_port//%/\\x5}")
+		echo "<p>Changing SSH port to: $ssh_port</p>"
+		rewrite_config /system/sdcard/config/ssh.conf ssh_port "$ssh_port"
+	fi
+  if [ -n ${F_ssh_key} ]; then
+		ssh_key=$(printf '%b' "${F_ssh_key//%/\\x}")
```
Comment (timstanley1985) [inline firmware_mod/www/cgi-bin/ui_system.cgi:90]: The spaces in the key get swapped for +'s for example "my+test+key"

File: firmware_mod/www/cgi-bin/ui_system.cgi [hunk 3]
```diff
@@ -78,6 +81,16 @@ if [ -n "$F_cmd" ]; then
 		echo "<p>Setting wifi password to: $wifi_password</p>"
 		wpa_config_set psk "\"$wifi_password\""
 	fi
+  if [ -n ${F_ssh_port} ]; then
+		ssh_port=$(printf '%b' "${F_ssh_port//%/\\x5}")
```
Comment (fhl206) [inline firmware_mod/www/cgi-bin/ui_system.cgi:85]:
```suggestion
		ssh_port=$(printf '%b' "${F_ssh_port//%/\\x}")
```

Comment (timstanley1985) [reply firmware_mod/www/cgi-bin/ui_system.cgi:85]: Yes, will fix

File: firmware_mod/www/cgi-bin/ui_system.cgi (hunk 2 above)
Comment (jmtatsch) [reply firmware_mod/www/cgi-bin/ui_system.cgi:90]: Plus signs in CGI parameters mean spaces. Apparently that's in the CGI spec.
Maybe run javascript to encode the spaces to %20 before sending to the key to the cgi script.
Then in the shell script use regex to replace to decode the spaces to real spaces again.

Comment (timstanley1985) [reply firmware_mod/www/cgi-bin/ui_system.cgi:90]: Thanks, that worked perfectly.  Will submit a final version of this patch tomorrow

**Comment by timstanley1985** [comment]
Thanks for your help, I think this is ready to merge unless you have any other comments

**Pull request merged**